protocol/src/main/java/org/fido/iot/protocol/To2ClientService.java

```diff
@@ -298,25 +299,48 @@ protected void doSetupDevice(Composite request, Composite reply) {
       newHash = getCryptoService().hash(hashType, secret, headerCopy.toBytes());
     }
 
-    Composite payload = Composite.newArray()
-        .set(Const.FIRST_KEY,
-            newHash != null ? newHash : PrimitivesUtil.getCborNullBytes());
+    Composite payload = Composite.newArray();
+    payload.set(
+        Const.FIRST_KEY,
+        newHash != null
+            ? newHash
+            : Composite.newArray().set(Const.FIRST_KEY, PrimitivesUtil.getCborNullBytes()));
+    payload.set(
+        Const.SECOND_KEY,
+        getStorage().getMaxOwnerServiceInfoMtuSz() != null
+            ? Integer.parseInt(getStorage().getMaxOwnerServiceInfoMtuSz())
+            : Composite.newArray().set(Const.FIRST_KEY, PrimitivesUtil.getCborNullBytes()));
 
     body = getCryptoService().encrypt(payload.toBytes(), this.ownState);
 
-    reply.set(Const.SM_MSG_ID, Const.TO2_AUTH_DONE);
+    reply.set(Const.SM_MSG_ID, Const.TO2_DEVICE_SERVICE_INFO_READY);
     reply.set(Const.SM_BODY, body);
     getStorage().continued(request, reply);
   }
 
-  protected void doAuthDone2(Composite request, Composite reply) {
+  protected void doOwnerServiceInfoReady(Composite request, Composite reply) {
     getStorage().continuing(request, reply);
     Composite body = request.getAsComposite(Const.SM_BODY);
     Composite message = Composite.fromObject(getCryptoService().decrypt(body, this.ownState));
 
-    message.verifyMaxKey(Const.NO_KEYS);
+    int deviceMtu = 0;
```

Comment:
Will this be a default of 0 if nothing specified in the message? Don't we want the default of 1300?

Reply:
Updated variable initialization to Default MTU constant. Not sure what should be the value if message is empty because spec says it should be 1300 bytes when null value is received as MTU size.